src/messaging/ExchangeDelegate.h

```diff
@@ -59,6 +56,10 @@ class DLL_EXPORT ExchangeDelegateBase
     virtual void OnMessageReceived(ExchangeContext * ec, const PacketHeader & packetHeader, const PayloadHeader & payloadHeader,
                                    System::PacketBufferHandle payload) = 0;
 
+    virtual void OnReceiveCredentials(SecureSessionHandle session, SecureSessionMgr * mgr, OperationalCredentialSet * opCredSet,
```

Comment:
I don't think `OnReceiveCredentials` should be a callback in the Exchange Delegate. The layer above should receive the message via `OnMessageReceived` and process it to figure out that it contains the credentials.